File: connectors/connector-jira/src/test/java/com/taskadapter/connector/jira/MockData.java
package com.taskadapter.connector.jira;

import com.atlassian.jira.rest.client.api.domain.BasicComponent;
import com.atlassian.jira.rest.client.api.domain.Field;
import com.atlassian.jira.rest.client.api.domain.Issue;
import com.atlassian.jira.rest.client.api.domain.IssueType;
import com.atlassian.jira.rest.client.api.domain.Priority;
import com.atlassian.jira.rest.client.api.domain.Version;
import com.atlassian.jira.rest.client.internal.json.FieldJsonParser;
import com.atlassian.jira.rest.client.internal.json.IssueJsonParser;
import com.atlassian.jira.rest.client.internal.json.JsonArrayParser;
import com.google.common.io.Resources;
import com.google.gson.reflect.TypeToken;
import com.taskadapter.connector.jira.mock.BasicComponentMock;
import com.taskadapter.connector.jira.mock.IssueTypeMock;
import com.taskadapter.connector.jira.mock.PriorityMock;
import com.taskadapter.connector.jira.mock.VersionMock;
import com.taskadapter.connector.testlib.TestDataLoader;
import org.assertj.core.util.Lists;
import org.codehaus.jettison.json.JSONObject;

import java.nio.charset.StandardCharsets;
import java.util.ArrayList;
import java.util.List;

public class MockData {
    public static Issue loadIssue(String fileName) {
        try {
            var issueParser = new IssueJsonParser();
            var fileContents = Resources.toString(Resources.getResource(fileName), StandardCharsets.UTF_8);
            var obj = new JSONObject(fileContents);
            return issueParser.parse(obj);
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }

    public static Iterable<Priority> loadPriorities() {
        return (Iterable<Priority>)
                TestDataLoader.load("priorities_6.4.11.json", new TypeToken<ArrayList<PriorityMock>>() {
                }.getType());
    }

    public static Iterable<IssueType> loadIssueTypes() {
        return (Iterable<IssueType>)
                TestDataLoader.load("issuetypes_jira5.0.6.json", new TypeToken<ArrayList<IssueTypeMock>>() {
                }.getType());
    }

    public static Iterable<Version> loadVersions() {
        /* I deleted "release date" attribute from the versions file to fix
         * "Unable to invoke no-args constructor for class org.joda.time.Chronology" problem.
         * See the original file "versions_jira5.0.6.json"
         */
        return (Iterable<Version>)
                TestDataLoader.load("versions_without_release_date_jira5.0.6.json", new TypeToken<ArrayList<VersionMock>>() {
                }.getType());
    }

    public static Iterable<BasicComponent> loadComponents() {
        return (Iterable<BasicComponent>)
                TestDataLoader.load("components_jira5.0.6.json", new TypeToken<ArrayList<BasicComponentMock>>() {
                }.getType());
    }

    public static Iterable<com.atlassian.jira.rest.client.api.domain.Field> loadFieldDefinitions() {
        try {
            var fieldsParser = FieldJsonParser.createFieldsArrayParser();
            Iterable<Field> fields = fieldsParser.parse(ResourceUtil.getJsonArrayFromResource("7.1.9/fields.json"));
            return Lists.newArrayList(fields);
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }
}

File: connectors/connector-jira/src/test/scala/com/taskadapter/connector/jira/MockData.scala
package com.taskadapter.connector.jira

import java.util

import com.atlassian.jira.rest.client.api.domain.{BasicComponent, Field, Issue, IssueType, Priority, Version}
import com.atlassian.jira.rest.client.internal.json.{FieldJsonParser, IssueJsonParser, JsonArrayParser}
import com.google.common.base.Charsets
import com.google.common.io.Resources
import com.google.gson.reflect.TypeToken
import com.taskadapter.connector.jira.mock._
import com.taskadapter.connector.testlib.TestDataLoader
import org.codehaus.jettison.json.JSONObject
import scala.collection.JavaConverters._

object MockData {
  def loadIssue(fileName: String): Issue = {
    val issueParser = new IssueJsonParser
    val fileContents = Resources.toString(Resources.getResource(fileName), Charsets.UTF_8)
    val obj = new JSONObject(fileContents)
    issueParser.parse(obj)
  }

  def loadPriorities: Iterable[Priority] = TestDataLoader.load("priorities_6.4.11.json",
    new TypeToken[util.ArrayList[PriorityMock]]() {}.getType).asInstanceOf[java.lang.Iterable[Priority]]
    .asScala

  def loadIssueTypes: Iterable[IssueType] = TestDataLoader.load("issuetypes_jira5.0.6.json",
    new TypeToken[util.ArrayList[IssueTypeMock]]() {}.getType).asInstanceOf[java.lang.Iterable[IssueType]]
    .asScala

  def loadFieldDefinitions: Iterable[Field] = {
    val fieldsParser: JsonArrayParser[java.lang.Iterable[Field]] = FieldJsonParser.createFieldsArrayParser
    fieldsParser.parse(ResourceUtil.getJsonArrayFromResource("7.1.9/fields.json")).asScala
  }

  def loadVersions: Iterable[Version] = {
    /* I deleted "release date" attribute from the versions file to fix
     * "Unable to invoke no-args constructor for class org.joda.time.Chronology" problem.
     * See the original file "versions_jira5.0.6.json"
     */
    TestDataLoader.load("versions_without_release_date_jira5.0.6.json",
      new TypeToken[util.ArrayList[VersionMock]]() {}.getType).asInstanceOf[java.lang.Iterable[Version]]
      .asScala
  }

  def loadComponents: Iterable[BasicComponent] = TestDataLoader.load("components_jira5.0.6.json",
    new TypeToken[util.ArrayList[BasicComponentMock]]() {}.getType).asInstanceOf[java.lang.Iterable[BasicComponent]]
    .asScala
}
